cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewServer.java

```diff
@@ -0,0 +1,292 @@
+import co.cask.cdap.store.guice.NamespaceStoreModule;
+import co.cask.http.HandlerHook;
+import co.cask.http.HttpHandler;
+import co.cask.http.NettyHttpService;
+import co.cask.tephra.TransactionManager;
+import co.cask.tephra.inmemory.InMemoryTransactionService;
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.common.util.concurrent.Futures;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.TypeLiteral;
+import com.google.inject.name.Names;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.twill.common.Cancellable;
+import org.apache.twill.discovery.Discoverable;
+import org.apache.twill.discovery.DiscoveryService;
+import org.apache.twill.discovery.InMemoryDiscoveryService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.util.Set;
+
+/**
+ * Preview Server.
+ */
+public class PreviewServer extends AbstractIdleService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(PreviewServer.class);
+
+  private final CConfiguration cConf;
+  private final DiscoveryService discoveryService;
+  private final ProgramRuntimeService programRuntimeService;
+  private final ApplicationLifecycleService applicationLifecycleService;
+  private final ProgramLifecycleService programLifecycleService;
+  private final SystemArtifactLoader systemArtifactLoader;
+
+  private final MetricsCollectionService metricsCollectionService;
+  private final LogAppenderInitializer logAppenderInitializer;
+  private final DatasetService datasetService;
+
+  private final NettyHttpService httpService;
+  private Cancellable cancellable;
+
+  /**
+   * Creates Preview Injector based on shared bindings passed from Standalone injector and
+   * constructs the PreviewServer with configuration coming from guice injection.
+   */
+  @Inject
+  public PreviewServer(CConfiguration cConf, Configuration hConf,
+                       InMemoryDiscoveryService discoveryService, TransactionManager transactionManager,
+                       DatasetFramework datasetFramework, final ArtifactRepository artifactRepository,
+                       final ArtifactStore artifactStore, final AuthorizerInstantiator authorizerInstantiator,
+                       final StreamAdmin streamAdmin, final StreamConsumerFactory streamConsumerFactory,
+                       final StreamCoordinatorClient streamCoordinatorClient,
+                       final PrivilegesManager privilegesManager,
+                       final AuthorizationEnforcer authorizationEnforcer,
+                       final InMemoryTransactionService inMemoryTransactionService) {
```

Comment:
This constructor is called directly by standalone,its not injected. I kept the inject tag so we don't need a default constructor. should i change that?